src/ZODB/FileStorage/tests.py

```diff
@@ -185,6 +185,125 @@ def _save_index():
 
     """
 
+def pack_disk_full_copyToPacktime():
+    """Recover from a disk full situation by removing the `.pack` file
+
+`copyToPacktime` fails
+
+Add some data
+
+    >>> fs = ZODB.FileStorage.FileStorage('data.fs')
+    >>> db = ZODB.DB(fs)
+    >>> conn = db.open()
+    >>> conn.root()[1] = u'foobar'
+    >>> transaction.commit()
+
+patch `copyToPacktime` to fail
+
+    >>> from ZODB.FileStorage import fspack
+    >>> save_copyToPacktime = fspack.FileStoragePacker.copyToPacktime
+
+    >>> def failing_copyToPacktime(self):
+    ...     self._tfile.write('somejunkdata')
```

Comment:
`_tfile` is [opened in binary mode](https://github.com/zopefoundation/ZODB/blob/115158a3fc121c40709592f81152ec3b1b701243/src/ZODB/FileStorage/fspack.py#L412), so it needs bytes:

```python
self._tfile.write(b'somejunkdata')
```